src/@next/CurrencyInput/CurrencyInput.tsx

```diff
@@ -35,25 +35,31 @@ export const CurrencyInput = React.forwardRef<
     console.warn(`Locale value of ${locale} is unsupported, "en" will be used`);
   }
 
-  const formatter = new Intl.NumberFormat(localeValue);
+  const formatter = useMemo(
```

Comment:
It becomes a chaining effect, it gives another warning that I should wrap this with useMemo because it is used inside a useCallback